File: src/python/test_sqlite.py
#!/usr/bin/env python3
"""
Test de conectividad Python <-> SQLite (Contrato Fase 0)
"""
import sqlite3
import os

def test_python_sqlite():
    print("========================================")
    print("[Python] Test de Conectividad SQLite3")
    print("========================================")
    
    db_path = os.path.join("data", "pea_investigacion.db")
    if not os.path.exists(db_path):
        print(f"[ERROR Python] No se encontró el archivo: {db_path}")
        return False
        
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    print(f"[OK Python] Base de datos conectada exitosamente: {db_path}\n")
    
    # 1. Conteo de Grupos
    cursor.execute("SELECT COUNT(*) FROM Grupos WHERE activo = 1;")
    num_grupos = cursor.fetchone()[0]
    
    # 2. Conteo de Investigadores
    cursor.execute("SELECT COUNT(*) FROM Investigadores WHERE activo = 1;")
    num_inv = cursor.fetchone()[0]
    
    # 3. Conteo de Productos
    cursor.execute("SELECT COUNT(*) FROM Productos WHERE activo = 1;")
    num_prod = cursor.fetchone()[0]
    
    print(f"• Total Grupos Activos:         {num_grupos}")
    print(f"• Total Investigadores Activos: {num_inv}")
    print(f"• Total Productos Activos:      {num_prod}")
    
    print("\n--- Investigadores y su Grupo ---")
    query = """
    SELECT i.nombre_completo, i.categoria, g.nombre 
    FROM Investigadores i
    JOIN Grupos g ON i.codigo_grupo = g.codigo_grupo
    WHERE i.activo = 1;
    """
    cursor.execute(query)
    for row in cursor.fetchall():
        print(f"• {row[0]} ({row[1]}) -> {row[2]}")
        
    conn.close()
    print("\n[OK Python] Prueba de persistencia finalizada exitosamente.")
    return True

if __name__ == "__main__":
    test_python_sqlite()

File: src/cpp/test_sqlite.cpp
#include <iostream>
#include <sqlite3.h>
#include <string>

// Test de conectividad C++ <-> SQLite (Contrato Fase 0)
int main() {
    std::cout << "========================================\n";
    std::cout << "[C++] Test de Conectividad SQLite3\n";
    std::cout << "========================================\n";

    sqlite3* db = nullptr;
    const std::string db_path = "data/pea_investigacion.db";

    int rc = sqlite3_open(db_path.c_str(), &db);
    if (rc != SQLITE_OK) {
        std::cerr << "[ERROR C++] No se pudo abrir la BD: " << sqlite3_errmsg(db) << "\n";
        return 1;
    }
    std::cout << "[OK C++] Base de datos abierta exitosamente: " << db_path << "\n\n";

    // Consulta de prueba a Grupos
    const char* sql = "SELECT codigo_grupo, nombre, clasificacion, anio_creacion FROM Grupos WHERE activo = 1;";
    sqlite3_stmt* stmt = nullptr;

    rc = sqlite3_prepare_v2(db, sql, -1, &stmt, nullptr);
    if (rc != SQLITE_OK) {
        std::cerr << "[ERROR C++] Error al preparar la consulta: " << sqlite3_errmsg(db) << "\n";
        sqlite3_close(db);
        return 1;
    }

    std::cout << "--- Grupos Registrados (Activos) ---\n";
    while (sqlite3_step(stmt) == SQLITE_ROW) {
        std::string codigo = reinterpret_cast<const char*>(sqlite3_column_text(stmt, 0));
        std::string nombre = reinterpret_cast<const char*>(sqlite3_column_text(stmt, 1));
        std::string clasif = reinterpret_cast<const char*>(sqlite3_column_text(stmt, 2));
        int anio = sqlite3_column_int(stmt, 3);

        std::cout << "• [" << codigo << "] " << nombre << " | Clasif: " << clasif << " | Creación: " << anio << "\n";
    }

    sqlite3_finalize(stmt);
    sqlite3_close(db);
    std::cout << "\n[OK C++] Prueba finalizada exitosamente sin fugas.\n";
    return 0;
}
